connect/src/main/java/com/stripe/android/connect/EmbeddedComponentManager.kt

```diff
@@ -154,42 +157,81 @@ class EmbeddedComponentManager(
             return true
         }
 
-        val launcher = getLauncher(context, requestPermissionLaunchers, "Error launching camera permission request")
-            ?: return null
+        val (_, launcher) =
+            getLauncher(context, requestPermissionLaunchers, "Error launching camera permission request")
+                ?: return null
         launcher.launch(Manifest.permission.CAMERA)
 
         return permissionsFlow.first()
     }
 
     internal suspend fun chooseFile(context: Context, requestIntent: Intent): Array<Uri>? {
-        val launcher = getLauncher(context, chooseFileLaunchers, "Error choosing file")
+        val (activity, launcher) = getLauncher(context, chooseFileLaunchers, "Error choosing file")
             ?: return null
         launcher.launch(requestIntent)
 
-        return chooseFileResultFlow.first()
+        return chooseFileResultFlow
+            .first { it.activity == activity }
```

Comment:
👀 Being extra careful to ensure activities don't get results from other activities.